syncopy/connectivity/ST_compRoutines.py

```diff
@@ -0,0 +1,428 @@
+        _, freq_idx = best_match(freqs, foi, squash_duplicates=True)
+        nFreq = freq_idx.size
+    else:
+        freq_idx = slice(None)
+        nFreq = freqs.size
+
+    # we always average over tapers here
+    outShape = (1, nFreq, nChannels, nChannels)
+
+    # For initialization of computational routine,
+    # just return output shape and dtype
+    # cross spectra are complex!
+    if noCompute:
+        return outShape, spectralDTypes["fourier"]
+
+    # detrend
+    if polyremoval == 0:
+        # SciPy's overwrite_data not working for type='constant' :/
+        dat = detrend(dat, type='constant', axis=0, overwrite_data=True)
+    elif polyremoval == 1:
+        dat = detrend(dat, type='linear', axis=0, overwrite_data=True)
+
+    # compute the individual spectra
+    # specs have shape (nTapers x nFreq x nChannels)
+    specs, freqs = mtmfft(dat, samplerate, taper, taper_opt)
+
+    # outer product along channel axes
+    # has shape (nTapers x nFreq x nChannels x nChannels)
+    CS_ij = specs[:, :, np.newaxis, :] * specs[:, :, :, np.newaxis].conj()
+
+    # average tapers and transpose:
+    # now has shape (nChannels x nChannels x nFreq)
+    CS_ij = CS_ij.mean(axis=0).T
+
+    if norm:
+        # only meaningful for multi-tapering
+        if taper != 'dpss':
+            msg = "Normalization of single trial csd only possible with taper='dpss'"
+            raise SPYValueError(legal=msg, varname="taper", actual=taper)
+        # main diagonal has shape (nChannels x nFreq): the auto spectra
+        diag = CS_ij.diagonal()
+        # get the needed product pairs of the autospectra
+        Ciijj = np.sqrt(diag[:, :, None] * diag[:, None, :]).T
+        CS_ij = CS_ij / Ciijj
+
+    # where does freqs go/come from -
+    # we will eventually allow tuples as return values yeah!
+    if not fullOutput:
+        return CS_ij[None, ..., freq_idx].transpose(0, 3, 1, 2)
+    else:
+        return CS_ij[None, ..., freq_idx].transpose(0, 3, 1, 2), freqs[freq_idx]
+
+
+class ST_CrossSpectra(ComputationalRoutine):
+
+    """
+    Compute class that calculates single-trial (multi-)tapered cross spectra
+    of :class:`~syncopy.AnalogData` objects
+
+    Sub-class of :class:`~syncopy.shared.computational_routine.ComputationalRoutine`,
+    see :doc:`/developer/compute_kernels` for technical details on Syncopy's compute
+    classes and metafunctions.
+
+    See also
+    --------
+    syncopy.connectivityanalysis : parent metafunction
+    """
+
+    # the hard wired dimord of the cF
+    dimord = ['time', 'freq', 'channel_i', 'channel_j']
+
+    computeFunction = staticmethod(cross_spectra_cF)
+
+    backends = [mtmfft]
+    # 1st argument,the data, gets omitted    
+    valid_kws = list(signature(mtmfft).parameters.keys())[1:]
+    valid_kws += list(signature(cross_spectra_cF).parameters.keys())[1:]
+    # hardcode some parameter names which got digested from the frontend
+    valid_kws += ['tapsmofrq', 'nTaper']    
```

Comment:
I still think it's not suuuper neat, due to the possible discrepancy between kwarg names passed to the frontend vs. the kwargs passed to the CRs, hence this kind of monkey patching here.. but I hope this gives a bit more context to the specific CRs, and so we can use the `check_effective_parameters()` instead of having to worry about combinatorial crazyness of parameter combinations